core/storage/suggestion/gae_models_test.py

```diff
@@ -0,0 +1,209 @@
+            suggestion_models.SUGGESTION_EDIT_STATE_CONTENT,
+            suggestion_models.TARGET_TYPE_EXPLORATION,
+            self.target_id, self.target_version_at_submission,
+            suggestion_models.STATUS_REJECTED, 'author_2',
+            'reviewer_2', 'reviewer_3', self.change_cmd, self.score_category,
+            'thread_4')
+        suggestion_models.SuggestionModel.create(
+            suggestion_models.SUGGESTION_EDIT_STATE_CONTENT,
+            suggestion_models.TARGET_TYPE_EXPLORATION,
+            self.target_id, self.target_version_at_submission,
+            suggestion_models.STATUS_REJECTED, 'author_3',
+            'reviewer_3', 'reviewer_2', self.change_cmd, self.score_category,
+            'thread_5')
+
+    def test_target_type_choices_contains_delimiter(self):
+        for target_type in suggestion_models.TARGET_TYPE_CHOICES:
+            self.assertTrue(
+                suggestion_models.SUGGESTION_ID_DELIMITER not in target_type)
+
+    def test_score_type_contains_delimiter(self):
+        for score_type in suggestion_models.SCORE_TYPE_CHOICES:
+            self.assertTrue(
+                suggestion_models.SCORE_CATEGORY_DELIMITER not in score_type)
+
+    def test_create_new_object_succesfully(self):
+        suggestion_models.SuggestionModel.create(
+            suggestion_models.SUGGESTION_EDIT_STATE_CONTENT,
+            suggestion_models.TARGET_TYPE_EXPLORATION,
+            self.target_id, self.target_version_at_submission,
+            suggestion_models.STATUS_IN_REVIEW, 'author_3',
+            'reviewer_3', 'reviewer_3', self.change_cmd, self.score_category,
+            'thread_6')
+
+        suggestion_id = suggestion_models.SuggestionModel.get_instance_id(
+            suggestion_models.TARGET_TYPE_EXPLORATION,
+            self.target_id, 'thread_6')
+
+        observed_suggestion = suggestion_models.SuggestionModel.get_by_id(
+            suggestion_id)
+
+
+        self.assertEqual(
+            observed_suggestion.suggestion_type,
+            suggestion_models.SUGGESTION_EDIT_STATE_CONTENT)
+        self.assertEqual(
+            observed_suggestion.target_type,
+            suggestion_models.TARGET_TYPE_EXPLORATION)
+        self.assertEqual(
+            observed_suggestion.target_id, self.target_id)
+        self.assertEqual(
+            observed_suggestion.target_version_at_submission,
+            self.target_version_at_submission)
+        self.assertEqual(
+            observed_suggestion.status, suggestion_models.STATUS_IN_REVIEW)
+        self.assertEqual(observed_suggestion.author_id, 'author_3')
+        self.assertEqual(observed_suggestion.assigned_reviewer_id, 'reviewer_3')
+        self.assertEqual(observed_suggestion.reviewer_id, 'reviewer_3')
+        self.assertEqual(
+            observed_suggestion.score_category, self.score_category)
+        self.assertEqual(observed_suggestion.change_cmd, self.change_cmd)
+
+    def test_fail_create_object(self):
```

Comment:
test_create_suggestion_fails_if_id_collides_with_existing_one()

(or similar. Ideally the developer can understand the test from the name without having to read it.)

Reply:
Done.